internal/static/parse.go

```diff
@@ -0,0 +1,72 @@
+package static
+
+import (
+	"go/constant"
+	"go/types"
+
+	"github.com/dogmatiq/configkit"
+	"github.com/dogmatiq/configkit/internal/entity"
+	"github.com/dogmatiq/configkit/internal/typename/gotypes"
+	"golang.org/x/tools/go/ssa"
+)
+
+// parse parses the member of the package and populates it into
+// entity.Application.
+//
+// The second parameter is the type that actually implements
+// github.com/dogmatiq/dogma.Application interface.
+func parse(m ssa.Member, typ types.Type) *entity.Application {
+	app := &entity.Application{
+		TypeNameValue: gotypes.NameOf(typ),
+	}
+
+	fn := m.Package().Prog.LookupMethod(
+		typ,
+		m.Package().Pkg,
+		"Configure",
+	)
+
+	// Since `.Configure()` is the method, the first parameter is the receiver,
+	// so we need to extract the second paramenter as
+	// `dogma.ApplicationConfigurer`
+	configurer := fn.Params[1]
+
+	for _, c := range filterCalls(fn) {
+		// Check if we are calling the methods on the configurer.
+		if c.Common().Value.Name() == configurer.Name() {
```

Comment:
There's probably an opportunity here to make a more specific function than `filterCalls()`, like perhaps you could wrap a lot of this up in a function like `findConfigurerCalls(fn)` that would be usable for all of the handler types as well (since the configurer is always the first and only parameter to the Configure method).

There's nothing really wrong with how it is now, but since you are iterating over everything to find the `ssa.Call` values we may as well find the exact ones we need in that single iteration.

Reply:
This hopefully is addressed in https://github.com/dogmatiq/configkit/pull/118/commits/b4777f379b5ec0bd39486dad955fee964c8428c1.